core/src/main/java/io/questdb/cairo/map/CompactMap.java

```diff
@@ -670,27 +675,24 @@ private void putEntryAt(long entryOffset, long slot, byte flag) {
         }
 
         private CompactMapValue putNewEntryAt(long slot, byte flag) {
-            if (size < maxSize) {
-                // entry size is now known
-                // values are always fixed size and already accounted for
-                // so go ahead and finalize
-                entries.putByte(currentEntryOffset, flag);
-                entries.putLong(currentEntryOffset + 1, currentEntrySize); // size
-                entries.jumpTo(currentEntryOffset + ENTRY_HEADER_SIZE);
-
-                if (++size == keyCapacity) {
-                    // reached capacity?
-                    // no need to populate slot, grow() will do the job for us
-                    grow();
-                } else {
-                    setOffsetAt(slot, currentEntryOffset);
-                }
-                // this would be offset of entry values
-                value.of(currentEntryOffset, true);
-                return value;
+            resourceLimiter.checkLimits(size);
```

Comment:
both maps are fixed size structure most of the time, checking limits should be done on resize. We can perhaps skip initial sizing but prohibit further resize if limits do no allow.

Also limit check call will be inside hopefully non-inlined call to grow() and have 0 performance impact on most things